macosx/kext/AudioEngine.cpp

```diff
@@ -651,16 +706,36 @@ IOReturn kXAudioEngine::performFormatChange(IOAudioStream *audioStream, const IO
     
     if (newSampleRate)
     {
-        switch (newSampleRate->whole) {
-            case 48000:
-                debug("\t-> 48kHz selected\n");
-                
-					// Add code to switch hardware to 48kHz
-                break;
-            default:
-					// This should not be possible since we only specified 44100 and 48000 as valid sample rates
-                debug("\t Internal Error - unknown sample rate selected.\n");
-                break;
+        if(!hw->is_edsp){
+            for(int i=0; i<n_channels;i++)
+            {
+                hw->voicetable[i].param.initial_pitch =(word)  kx_srToPitch(kx_sr_coeff(hw,newSampleRate->whole) >> 8);
```

Comment:
While this adds support for other sampling rates, in reality it produces worse audio quality output.
Hardware SRC on 10k1/10k2 devices was great 15 years ago, but modern software is better :)

Reply:
At any time can choose 48kHz.